ministryofjustice/laa-crime-application-store
# CRM457-1112 Remove Authentication for local development

## Description of change
Have overridden the creation of `SingleTenantAzureAuthorizationCodeBearer` so that it will be ignored if the environment variable `AUTHENTICATION_REQUIRED` is set to anything other than True

Have also replaced start_up events with lifespan as they have been deprecated.

New tests to cover the unauthenticated situation

Updated envVars so can have different requirements for testing than dev if required.

## Link to relevant ticket
[CRM457-1152](https://dsdmoj.atlassian.net/browse/CRM457-1152)


## How to manually test the feature
Download this branch and point a provider application at it. This should work without specifying azure authentication tokens in the provider application.

You should also look to try and see if authentication works locally if environment variable set to True

[CRM457-1152]: https://dsdmoj.atlassian.net/browse/CRM457-1152?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
